bowling/src/frameParser.cpp

```diff
@@ -0,0 +1,40 @@
+#include <algorithm>
+#include <functional>
+
+#include "frameParser.hpp"
+#include "tokenizer.hpp"
+
+const char FrameParser::StrikeSign = 'X';
+const char FrameParser::MissSign = '-';
+const char FrameParser::SpareSign = '/';
+
+
+size_t FrameParser::parseToken(const char token, size_t& lastParsedToken) {
```

Comment:
To be honest, I have also feeling that it is not the best way to do that. I have tried with lambda function and capture list, but this was also derided in previous review. I do not really have idea what to do. I will need this last parsed element anyway. 

To the first comment - please note that is actually "calculated" value (integer/size_t) of token 10, 5 ,3 or so, not token itself '/' '3'.

Reply:
I will leave as it is (maybe there is obvoius way for doing this - there kind of problems are probably common).

Reply:
> To the first comment - please note that is actually "calculated" value (integer/size_t) of token 10, 5 ,3 or so, not token itself '/' '3'.
Then maybe `lastRollValue` or just `lastValue` ?

Reply:
And related to other stuff:

I've tried this:
```cpp
void FrameParser::parseToken(const char token, size_t& lastParsedToken) {
    switch (token) {
    case FrameParser::StrikeSign:
        lastParsedToken = 10;
        break;
    case FrameParser::SpareSign:
        lastParsedToken = 10 - lastParsedToken;
        break;
    case FrameParser::MissSign:
        lastParsedToken = 0;
        break;
    default:
        lastParsedToken = static_cast<size_t>(token - '0');
        break;
    }
}
```
As you can see I've changed return type to `void`.

And in `FrameParser::parse()` method, instead of `std::transform` I've called this:
```cpp
    std::for_each(sequence.cbegin(), sequence.cend(),
                  [&lastParsedToken, &parsed](const auto& el){
                      FrameParser::parseToken(el, lastParsedToken);
                      parsed.push_back(lastParsedToken);
                  });
```

Just my suggestion, not the ultimate solution.  ;) But tests are still fine.

Reply:
Done.